kernel/CMakeLists.txt

```diff
@@ -59,6 +59,7 @@ list(APPEND kernel_files
   mem_slab.c
   thread.c
   version.c
+  sched.c
```

Comment:
Strangely, there are some `#ifdef CONFIG_MULTITHREADING` items in `sched.c`, so it looks like (at one point) it was already being compiled in single-threaded mode.

However, I have some concerns that there still might be some things that should not be compiled in `sched.c` when `CONFIG_MULTITHREADING=n`.

For example, the whole `struct _thread_base base` field inside of `struct k_thread`, and all of the functions in `sched.c` that currently touch that field.